pkg/volume/host_path/host_path.go

```diff
@@ -123,8 +123,12 @@ func (plugin *hostPathPlugin) NewUnmounter(volName string, podUID types.UID) (vo
 	}}, nil
 }
 
-func (plugin *hostPathPlugin) NewRecycler(pvName string, spec *volume.Spec) (volume.Recycler, error) {
-	return plugin.newRecyclerFunc(pvName, spec, plugin.host, plugin.config)
+func (plugin *hostPathPlugin) Recycle(pvName string, spec *volume.Spec) error {
+	recycler, err := plugin.newRecyclerFunc(pvName, spec, plugin.host, plugin.config)
```

Comment:
I will check again when I come back from holidays next week if I do not get time before.
But the main reason was, by copying the content, tests were failing because the tests were mocking the behaviour of the recycle.